### PR TITLE
Docs for ALTER ORDER BY

### PR DESCRIPTION
**Pull request by ztlpn**

### CODE REVIEW

File: docs/en/operations/table_engines/mergetree.md [hunk 1]
```diff
@@ -50,11 +51,13 @@ For a description of request parameters, see [request description](../../query_l
 
 - `ENGINE` - Name and parameters of the engine. `ENGINE = MergeTree()`. `MergeTree` engine does not have parameters.
 
-- `ORDER BY` — Primary key.
+- `ORDER BY` — Primary key (or sorting key if the separate `PRIMARY KEY` clause is present).
 
     A tuple of columns or arbitrary expressions. Example: `ORDER BY (CounterID, EventDate)`.
 If a sampling expression is used, the primary key must contain it. Example: `ORDER BY (CounterID, EventDate, intHash32(UserID))`.
 
+- `PRIMARY KEY` - Primary key if it differs from the [sorting key](mergetree.md#table_engines-mergetree-sorting_key) (the sorting key in this case is specified by the `ORDER BY` clause).
```
Comment (blinkov) [inline docs/en/operations/table_engines/mergetree.md]: I think it'd be better to break the cyclic dependency and just say that 1) by default primary key is equal to sorting key 2) few lines above ORDER KEY is just a sorting key without forward reference to PRIMARY KEY.

File: docs/en/operations/table_engines/mergetree.md [hunk 2]
```diff
@@ -161,10 +164,33 @@ The number of columns in the primary key is not explicitly limited. Depending on
 
 - Provide additional logic when data parts merging in the [CollapsingMergeTree](collapsingmergetree.md#table_engine-collapsingmergetree) and [SummingMergeTree](summingmergetree.md#table_engine-summingmergetree) engines.
 
-    You may need many fields in the primary key even if they are not necessary for the previous steps.
+    If you need this, it makes sense to specify the *sorting key* that is distinct from the primary key.
```
Comment (blinkov) [inline docs/en/operations/table_engines/mergetree.md]: If you need this -> In some cases ?

File: docs/en/operations/table_engines/mergetree.md [hunk 3]
```diff
@@ -161,10 +164,33 @@ The number of columns in the primary key is not explicitly limited. Depending on
...
 
 A long primary key will negatively affect the insert performance and memory consumption, but extra columns in the primary key do not affect ClickHouse performance during `SELECT` queries.
 
+<a name="table_engines-mergetree-sorting_key"></a>
+
+### Choosing the Sorting Key that is distinct from the Primary Key
```
Comment (blinkov) [inline docs/en/operations/table_engines/mergetree.md]: "distinct" might get confused with keyword, I'd just say "...Sorting Key that Differs from The Primary Key"

File: docs/en/query_language/alter.md [hunk 4]
```diff
@@ -68,9 +68,28 @@ For tables that don't store data themselves (such as `Merge` and `Distributed`),
...
+
+### Manipulations With Key Expressions
+
+The following command is supported:
+
+``` sql
+MODIFY ORDER BY new_expression
+```
+
+It only works for tables in the `MergeTree` family (including replicated tables). The command changes the
```
Comment (blinkov) [inline docs/en/query_language/alter.md]: Make links from "MergeTree" and maybe "replicated"?

File: docs/en/query_language/alter.md [hunk 5]
```diff
@@ -68,9 +68,28 @@ For tables that don't store data themselves (such as `Merge` and `Distributed`),
...
+[sorting key](../operations/table_engines/mergetree.md#table_engines-mergetree-sorting_key) of the table
+to `new_expression` (an expression or a tuple of expressions). Primary key remains the same.
+
+The command is lightweight in a sense that it only changes metadata. To keep the property that data part
+rows are sorted by the sorting key expression you cannot add expressions containing existing columns
```
Comment (blinkov) [inline docs/en/query_language/alter.md]: "sorted by the sorting..." sounds bad, "ordered by the sorting..." would be better I think

File: docs/en/operations/table_engines/mergetree.md (hunk 2 above)
Comment (ztlpn) [reply docs/en/operations/table_engines/mergetree.md]: This comment applies specifically to the case when the user needs additional merge logic (specified in the bullet point above).

**Comment by ztlpn** [comment]
@blinkov Thanks for the comments. https://github.com/yandex/ClickHouse/commit/64bdb2a4040a6a10ce311d16e8553dcfc210cdf3

**Pull request merged**